builders/integrations/oracles/dia.md

```diff
@@ -0,0 +1,139 @@
+
+### Included Price Feeds
+
+[DIA/USD](https://diadata.org/app/price/asset/Ethereum/0x84cA8bc7997272c7CfB4D0Cd3D55cd942B3c9419/), [BTC/USD](https://diadata.org/app/price/asset/Bitcoin/0x0000000000000000000000000000000000000000/), [USDC/USD](https://diadata.org/app/price/asset/Ethereum/0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48/)
+
+## Supported token API endpoints
+
+DIA also supports API and GraphQL endpoints to return cryptocurrency price data. You can [visit the DIA Documentation](https://docs.diadata.org/products/token-price-feeds/access-api-endpoints) to see all API endpoints.
+
+# 🎨 NFT Floor Price Feeds
+
+DIA NFT floor price feeds provide smart contracts with real-time price information of [18,000+ NFT collections](https://diadata.org/app/floor-price), sourced on-chain with 100% transparency from [multiple, cross-chain NFT marketplaces](https://diadata.org/app/source/nft).
+
+## Supported NFT API endpoints
+
+DIA also supports API endpoints to return cryptocurrency price data. Developers can directly access the example endpoints listed below or [visit the DIA Documentation](https://docs.diadata.org/products/nft-floor-price-feeds/access-api-endpoints) to see all API endpoints.
+
+# 🎲 Random Number Generation
+DIA xRandom provides smart contracts with unpredictable and unbiased random numbers, facilitating the development of on-chain use cases such as lotteries, prediction markets, NFT launches and more.
+## Understanding Randomness
+DIA leverages the Drand public randomness beacon, and updates its oracle with round numbers, randomness and a signature. Drand runs distributed nodes to produce their randomness beacon. Drand uses [Pedersen's DKG (Distributed Key Generation) protocol](https://drand.love/docs/cryptography/#distributed-key-generation-dkg) to create collective private/public keys. Participants in their League of Entropy then generate randomness in rounds and broadcast it together with its signature.
+To learn more about Drand’s randomness beacon, watch the [following DIA tutorial](https://youtu.be/7HALDJr8V3g) and read [Drand’s documentation](https://drand.love/docs/overview/#how-drand-works).
+## How to use the DIA Randomness Oracle
+Anyone can access published random values via round ID.
+```
+{
+	"round": 1597683,
+	"randomness": "24138936fcbf7fc3951c928158be6998cee3af622142d0790333608d17a5c5f6",
+	"signature": "8c04905c0adf34f1fb007915d9ccc7d07b97305fc63952726f9367c87f36ab687c5e190c151f6ac4d760a9e009fc54230adb8513885449d649a229bc727be9ff347bdbce1c609cebf993b6ae57133fbcf23f96b15dbd3510cb5f2ade6b30b647",
+	"previous_signature": "ada42197a2db89866da4c44348f77f7868e41e961ec32e636b912d43c625386afae9e54944ac573047dbd227ee495b52059586c8d8cd0edfe18cc15ca0666a66651da1d62b12af2d0fac19735bed9298690a593571965c3ad7c7b11947e76ec0"
+}
+```
+
+
+The DIA randomness smart contract is structured as follows
```

Comment:
I checked out your docs, looks like you have a demo randomness contract deployed to Moonbase Alpha ([`0x48d351ab7f8646239bbade95c3cc6de3ef4a6cec`](https://moonbase.moonscan.io/address/0x48d351ab7f8646239bbade95c3cc6de3ef4a6cec)). Do you by chance have one for Moonriver or Moonbeam?

Reply:
We don't, for the same reason as before. But again, something that could be set up if required :) Let me know!